presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -317,7 +317,12 @@ public PlanOptimizers(
                                 new TransformCorrelatedScalarSubquery(), // must be run after TransformCorrelatedScalarAggregationToJoin
                                 new TransformCorrelatedLateralJoinToJoin(),
                                 new ImplementFilteredAggregations())),
-                new TransformCorrelatedSingleRowSubqueryToProject(),
+                new IterativeOptimizer(
```

Comment:
I do not see legacy role injected here

Reply:
> Do we want to have the legacy rule run in the optimizer?

Yes, we do. Otherwise there is no point to have it in the codebase. We want to run legacy optimizer when iterative optimizer is disabled. Notice that `IterativeOptimizer` accepts a list of legacy optimizers and a set of `rule`s in one of its constructors. When iterative optimizer is disabled then it run legacy rules, otherwise it runs new rules.

I propose you to read carefully how `IterativeOptimizer` is implemented as it is quite important class in Presto optimizer related codebase.

Reply:
@kokosing Did the same, however, I do not see an example of the constructor being used in ``PlanOptimizers``. Could you please point me to one?

Reply:
It is because we recently we removed them. We were migrating a lot of rules at Teradata, now we do not have so much power to continue on that.

See:
 - a7463e6d96ab417466a29c67a1f33b8b493ab600
 - 9f95e662510606ee383f4c2469dae10aaddb1e45
 - 1816553e6bccaf5bf1a5198ee3269347086fdcb2